src/connection.rs

```diff
@@ -188,6 +200,24 @@ impl Connection {
 
         Ok(())
     }
+
+    fn poll_message(&mut self, cx: &mut Context<'_>) -> Poll<Option<SubstreamMessage>> {
+        loop {
+            // filter out any substreams that have timed out
+            for (substream_id, instant) in self.pending_substreams.iter() {
+                if instant.elapsed() >= self.substream_timeout {
+                    return Poll::Ready(Some(SubstreamMessage {
+                        substream_id: substream_id.clone(),
+                        message_type: SubstreamMessageType::OpenTimeout,
```

Comment:
this shouldn't be a message type as it's never sent over the network - I would just put this loop into `poll`, or just have it in its own function that just clears out timed out substream data

Reply:
this should just directly be a call to `self.handle_timeout()`, doesn't need to return here